java/vector/src/main/codegen/templates/AbstractPromotableFieldWriter.java

```diff
@@ -42,7 +42,7 @@
    * @param type
    * @return
    */
-  abstract protected FieldWriter getWriter(MinorType type);
+  abstract protected FieldWriter getWriter(ArrowType type);
```

Comment:
Agreed that we need a more flexible solution. A Minor type for every list size is very inconvenient.
Regarding perf: There are a few places were we convert between minorType and arrowType each way. We also use ArrowType.equals instead of == between MinorTypes in getWriter(ArrowType) which is called in startList() which is called for each list creation. Just saying that we need to actually measure the consequence of this. I'm not trying to make you do more work than necessary. Separating the refactoring from the new type and vector allows to get the later merged while we finalize the former.